packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -1,22 +1,66 @@
 'use strict';
 
 const get = require('lodash/get');
+const semver = require('semver');
+
 const { parseSQSMessageBody, sendSQSMessage } = require('@cumulus/aws-client/SQS');
 const log = require('@cumulus/common/log');
-const { getMessageExecutionArn } = require('@cumulus/message/Executions');
+const {
+  getKnexClient,
+  Executions,
+} = require('@cumulus/db');
+const {
+  getMessageExecutionArn,
+  getMessageExecutionParentArn,
+  getMessageCumulusVersion,
+} = require('@cumulus/message/Executions');
 const Execution = require('../models/executions');
 const Granule = require('../models/granules');
 const Pdr = require('../models/pdrs');
 const { getCumulusMessageFromExecutionEvent } = require('../lib/cwSfExecutionEventUtils');
 
-const saveExecutionToDb = async (cumulusMessage) => {
+const isPostRDSDeploymentExecution = (cumulusMessage) => {
+  const minimumSupportedRDSVersion = process.env.RDS_DEPLOYMENT_CUMULUS_VERSION;
+  if (!minimumSupportedRDSVersion) {
+    throw new Error('RDS_DEPLOYMENT_CUMULUS_VERSION environment variable must be set');
+  }
+  const cumulusVersion = getMessageCumulusVersion(cumulusMessage);
+  return cumulusVersion
+    ? semver.gte(cumulusVersion, minimumSupportedRDSVersion)
+    : false;
+};
+
+const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
+  const executionIsPostDeployment = isPostRDSDeploymentExecution(cumulusMessage);
+  const parentArn = getMessageExecutionParentArn(cumulusMessage);
+  if (!executionIsPostDeployment || !parentArn) return executionIsPostDeployment;
+  return Executions.doesExecutionExist({
```

Comment:
It seemed to me that we only need to check if the immediate parent execution exists in RDS, since that's all the current execution write would need to refer to? Each child execution checks for the existence of its own parent during write. This is another benefit of this approach vs traversing all the way to the initial parent execution.